maven-core/src/main/java/org/apache/maven/ReactorReader.java

```diff
@@ -69,8 +61,12 @@ class ReactorReader implements MavenWorkspaceReader {
 
     public static final String PROJECT_LOCAL_REPO = "project-local-repo";
 
-    private static final Collection<String> COMPILE_PHASE_TYPES =
-            Arrays.asList("jar", "ejb-client", "war", "rar", "ejb3", "par", "sar", "wsr", "har", "app-client");
+    private static final Set<String> COMPILE_PHASE_TYPES;
+
+    static {
+        COMPILE_PHASE_TYPES = new HashSet<>(
+                Arrays.asList("jar", "ejb-client", "war", "rar", "ejb3", "par", "sar", "wsr", "har", "app-client"));
+    }
```

Comment:
Where is the benefit?

Reply:
contains is faster on Set then List

Reply:
True, but does the type has to change to set from collection for this?

Reply:
https://gist.github.com/tbw777/716907b110df7eac1792de7148f8eb32

Reply:
I think what @michael-o meant is that we can keep the `Collection<String>` instead of changing the type to `Set<String>`:
```
private static final Collection<String> COMPILE_PHASE_TYPES =
             new HashSet<>(Arrays.asList("jar", "ejb-client", "war", "rar", "ejb3", "par", "sar", "wsr", "har", "app-client"));
```

Reply:
Correct.